src/packet/superpacket.cpp

```diff
@@ -19,30 +21,43 @@ void SuperPacket::print_packet(FILE *out) {
     fprintf(out, "}\n");
 }
 
-SuperPacket::SuperPacket(void *pkt, uint32_t max_payload_len) {
+SuperPacket::SuperPacket(void *pkt, uint32_t max_payload_len, Config *c) {
+    struct radiotap_header *radiotaph;
+    struct wlan_header * wlanh;
+
     struct ip *ipv4h;
     struct ether_header *eth;
 
+    this->config = c;
+
     parseable = true;
 
-    this->max_payload_len = max_payload_len;
-    eth = (struct ether_header *)pkt;
+    if (c->wireless == 1) {
+        radiotaph = (struct radiotap_header *) pkt;
+        radiotap_header.set_raw(radiotaph);
 
-    /* Check if packet has an ethernet header */
-    if ((ntohs(eth->ether_type) == ETHERTYPE_IP) ||
-        ((ntohs(eth->ether_type) == 0x86DD))) {
-        ethernet_header.set_raw(eth);
-        ipv4h = (struct ip *)&eth[1];
-    } else {
-        ipv4h = (struct ip *)pkt;
-    }
+        wlanh = (struct wlan_header *) &radiotaph[1]; 
```

Comment:
I wonder if we get the link layer then split into separate functions in general.